### PR TITLE
proj: Patch test OPEN_MAX limit assumption

### LINKED ISSUE #142875: proj.x86_64-darwin build failing
### Describe the bug
https://hydra.nixos.org/job/nixpkgs/trunk/proj.x86_64-darwin

The build is failing on darwin since the update to 8.1.1 (https://github.com/NixOS/nixpkgs/pull/138479)

### Notify maintainers

@dotlambda @vbgl

### PR DESCRIPTION
**Pull request by toonn**
###### Motivation for this change

One of the tests had a Linux-based assumption on the number of files that can be opened. I have submitted this patch upstream here, https://github.com/OSGeo/PROJ/pull/2934, and it's been merged but I don't expect to be able to update to a release before ZHF ends.

ZHF: #144627

Fixes #142875

This would obviate the need for #145147.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/proj/default.nix [hunk 1]
```diff
@@ -20,6 +21,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-Z2nruyowC3NG4Wb8AFBL0PME/zp9D7SwQdMSl6VjH/w=";
   };
 
+  patches = [ (fetchpatch {
+      url = "https://github.com/OSGeo/PROJ/commit/ac113a8898cded7f5359f1edd3abc17a78eee9b4.patch";
```
Comment (risicle) [inline pkgs/development/libraries/proj/default.nix:25]:
```suggestion
      name = "fix-capi-test-open-max-limit.patch";
      url = "https://github.com/OSGeo/PROJ/commit/ac113a8898cded7f5359f1edd3abc17a78eee9b4.patch";
```

**Comment by toonn** [comment]
I ran nixpkgs-review:

```
3 packages marked as broken and skipped:
python38Packages.pygmt python39Packages.basemap python39Packages.pygmt

23 packages failed to build:
gdal gmt gnudatalanguage gplates libLAS mapcache mapproxy mapserver merkaartor openorienteering-mapper postgis postgresql11Packages.postgis postgresql12Packages.postgis postgresql14Packages.postgis python38Packages.cartopy python38Packages.gdal python38Packages.osmnx python38Packages.worldengine python39Packages.cartopy python39Packages.osmnx python39Packages.worldengine sumo udig

17 packages built:
gdal_2 libgeotiff libspatialite octavePackages.octproj osm2pgsql proj python38Packages.basemap python38Packages.fiona python38Packages.geopandas python38Packages.owslib python38Packages.pyproj python38Packages.rasterio python39Packages.fiona python39Packages.geopandas python39Packages.owslib python39Packages.pyproj python39Packages.rasterio
```

File: pkgs/development/libraries/proj/default.nix (hunk 1 above)
Comment (toonn) [reply pkgs/development/libraries/proj/default.nix:25]: Thanks for the suggestion, I'll use the commit message as the name like format-patch would.

**Review by risicle — APPROVED** [review]
Builds & tests pass, macos 10.15, nixos x86_64 & aarch64 (emulated)

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 145371` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>grass</li>
    <li>pdal</li>
    <li>python38Packages.pygmt</li>
    <li>python39Packages.basemap</li>
    <li>python39Packages.pygmt</li>
    <li>saga</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>gnudatalanguage</li>
    <li>gplates</li>
    <li>merkaartor</li>
    <li>openorienteering-mapper</li>
  </ul>
</details>
<details>
  <summary>34 packages built:</summary>
  <ul>
    <li>gdal (python39Packages.gdal)</li>
    <li>gdal_2</li>
    <li>gmt</li>
    <li>libLAS</li>
    <li>libgeotiff</li>
    <li>libspatialite</li>
    <li>mapcache</li>
    <li>mapproxy</li>
    <li>mapserver</li>
    <li>osm2pgsql</li>
    <li>postgresql11Packages.postgis</li>
    <li>postgresql12Packages.postgis</li>
    <li>postgresqlPackages.postgis (postgresql13Packages.postgis)</li>
    <li>postgresql14Packages.postgis</li>
    <li>proj</li>
    <li>python38Packages.basemap</li>
    <li>python38Packages.cartopy</li>
    <li>python38Packages.fiona</li>
    <li>python38Packages.gdal</li>
    <li>python38Packages.geopandas</li>
    <li>python38Packages.osmnx</li>
    <li>python38Packages.owslib</li>
    <li>python38Packages.pyproj</li>
    <li>python38Packages.rasterio</li>
    <li>python38Packages.worldengine</li>
    <li>python39Packages.cartopy</li>
    <li>python39Packages.fiona</li>
    <li>python39Packages.geopandas</li>
    <li>python39Packages.osmnx</li>
    <li>python39Packages.owslib</li>
    <li>python39Packages.pyproj</li>
    <li>python39Packages.rasterio</li>
    <li>worldengine-cli (python39Packages.worldengine)</li>
    <li>udig</li>
  </ul>
</details>

**Comment by veprbl** [comment]
@toonn Very nice!

**Pull request merged**